src/lib.rs

```diff
@@ -1228,10 +1419,78 @@ console.log(a);
       None,
     );
     assert!(result.is_ok());
+    if let Module::Es(actual) = result.unwrap() {
+      assert_eq!(actual.dependencies.len(), 4);
+      assert_eq!(actual.specifier, specifier);
+      assert_eq!(actual.media_type, MediaType::TypeScript);
+    } else {
+      panic!("unexpected module type");
+    }
+  }
+
+  #[test]
+  fn test_parse_module_import_assertions() {
+    let specifier = ModuleSpecifier::parse("file:///a/test01.ts").unwrap();
+    let result = parse_module(
+      &specifier,
+      None,
+      Arc::new(
+        r#"
+    import a from "./a.json" assert { type: "json" };
+    await import("./b.json", { assert: { type: "json" } });
+    "#
+        .to_string(),
+      ),
```

Comment:
Would be great to have a test for `export ... from ... assert {type: "json"};` and `await import(...", { assert: { type: variable } })`, but it's not crucial to land this PR

Reply:
let me add those...